packages/airnode-deployer/src/infrastructure/index.ts

```diff
@@ -511,6 +512,79 @@ export type Deployment = {
   versions?: DeploymentVersion[];
 };
 
+// If deploymentId is provided tries fetch only that one deployment and finishes early when found
+async function fetchDeployments(cloudProvider: CloudProvider['type'], deploymentId?: string) {
+  const deployments: Deployment[] = [];
+
+  const bucketName = await cloudProviderLib[cloudProvider].getAirnodeBucket();
+  if (!bucketName) {
+    logger.debug(`No deployments available on ${cloudProvider.toUpperCase()}. Skipping.`);
+    return deployments;
+  }
+
+  const directoryStructure = await cloudProviderLib[cloudProvider].getBucketDirectoryStructure(bucketName);
+  for (const [airnodeAddress, addressDirectory] of Object.entries(directoryStructure)) {
+    if (addressDirectory.type !== FileSystemType.Directory) {
+      logger.warn(
+        `Invalid item in bucket '${bucketName}' (${cloudProvider.toUpperCase()}) with key '${
+          addressDirectory.bucketKey
+        }'. Skipping.`
+      );
+      continue;
+    }
+
+    for (const [stage, stageDirectory] of Object.entries(addressDirectory.children)) {
+      if (stageDirectory.type !== FileSystemType.Directory) {
+        logger.warn(
+          `Invalid item in bucket '${bucketName}' (${cloudProvider.toUpperCase()}) with key '${
+            stageDirectory.bucketKey
+          }'. Skipping.`
+        );
+        continue;
+      }
+
+      const latestDeployment = Object.keys(stageDirectory.children).sort().reverse()[0];
+      const bucketConfigPath = `${airnodeAddress}/${stage}/${latestDeployment}/config.json`;
+      const config = JSON.parse(
+        await cloudProviderLib[cloudProvider].getFileFromBucket(bucketName, bucketConfigPath)
+      ) as Config;
+      const region = (config.nodeSettings.cloudProvider as CloudProvider).region;
+      const airnodeVersion = config.nodeSettings.nodeVersion;
+      const id = hashDeployment(cloudProvider, region, airnodeAddress, stage, airnodeVersion);
+
+      const deploymentVersions = Object.keys(stageDirectory.children).map((versionTimestamp) => ({
+        id: hashDeploymentVersion(cloudProvider, region, airnodeAddress, stage, airnodeVersion, versionTimestamp),
+        timestamp: versionTimestamp,
+      }));
+      const deployment = {
+        id,
+        cloudProvider,
+        region,
+        airnodeAddress,
+        stage,
+        airnodeVersion,
+        lastUpdate: latestDeployment,
+        versions: deploymentVersions,
+      };
+
+      // We're looking for just one deployment
+      if (deploymentId) {
+        if (deploymentId === id) {
+          // Return the deployment if found
+          return [deployment];
+        } else {
```

Comment:
This is a bit awkward, why not just do it like this:

```
if (deploymentId === id) return [deployment]
deployments.push(deployment);
```

Reply:
I tried to explain in the comments why this won't work. Imagine that you've specified a `deploymentId` you're searching for but such a deployment doesn't exist. With your snippet, you would still fill up the `deployments` array with all the deployments and you wouldn't be able to tell that such a deployment was not found.

Reply:
But when you reach the return at the end, you know you haven't found it.

Reply:
Yes, and I would have to add another `if` statement there to see whether I'm looking for one specific deployment or not. And that's basically what I already have here.